django/river/api/views.py

```diff
@@ -0,0 +1,70 @@
+from inspect import getdoc, getmembers, isfunction, ismodule
+
+from rest_framework import generics, response, status, viewsets
+from rest_framework.decorators import action
+
+import scripts
+from river import models
+from river.adapters.event_publisher import KafkaEventPublisher
+from river.adapters.mappings import RedisMappingsRepository
+from river.adapters.topics import KafkaTopics
+from river.api import serializers
+from river.services import abort, batch, preview
+
+
+class BatchViewSet(viewsets.ModelViewSet):
+    queryset = models.Batch.objects.all()
+    serializer_class = serializers.BatchSerializer
+
+    def create(self, request):
+        serializer = serializers.BatchSerializer(data=request.data)
+        serializer.is_valid(raise_exception=True)
+        data = serializer.validated_data
+        resource_ids = data["resources"]
+
+        topics, event_publisher, mappings_repo = (
+            KafkaTopics(),
+            KafkaEventPublisher(),
+            RedisMappingsRepository(),
+        )
+        batch_instance = batch(resource_ids, topics, event_publisher, mappings_repo)
+
+        serializer = serializers.BatchSerializer(batch_instance)
+        return response.Response(serializer.data, status=status.HTTP_201_CREATED)
+
+    def destroy(self, request, *args, **kwargs):
+        batch_instance = self.get_object()
+
+        topics = KafkaTopics()
+        abort(batch_instance, topics)
+
+        return response.Response(status=status.HTTP_204_NO_CONTENT)
+
+    @action(methods=["post"], detail=True)
+    def retry(self, request, *args, **kwargs):
+        raise NotImplementedError
+
+
+class PreviewEndpoint(generics.CreateAPIView):
```

Comment:
I think some attributes are missing in this view (and it breaks the openapi schema generation). From the `GenericAPIView` doc:
```python
class GenericAPIView(views.APIView):
"""
    Base class for all other generic views.
    """
    # You'll need to either set these attributes,
    # or override `get_queryset()`/`get_serializer_class()`.
    # If you are overriding a view method, it is important that you call
    # `get_queryset()` instead of accessing the `queryset` property directly,
    # as `queryset` will get evaluated only once, and those results are cached
    # for all subsequent requests.
    queryset = None
    serializer_class = None
```